### ISSUE TITLE
"is_all_dates" inconsistent across datetimelike types

### ISSUE THREAD

**Issue by toobaz** [pandas-dev/pandas#19204]
xref #19187 

#### Code Sample, a copy-pastable example if possible

```
In [2]: pd.DatetimeIndex(['2000-02-01', '2000-02-02']).is_all_dates
Out[2]: True

In [3]: pd.DatetimeIndex(['2000-02-01', '2000-02-02']).append(pd.Index(['not a date']))[:2].is_all_dates
Out[3]: True

In [4]: pd.PeriodIndex(['2000', '2001'], freq='A').is_all_dates
Out[4]: True

In [5]: pd.PeriodIndex(['2000', '2001'], freq='A').append(pd.Index(['not a date']))[:2].is_all_dates
Out[5]: False

In [6]: pd.TimedeltaIndex([2000, 2001]).append(pd.Index(['not a date']))[:2].is_all_dates
Out[6]: False
```

#### Problem description

``is_all_dates`` is smart for ``object`` indexes but only for ``DatetimeIndex``, which is inconsistent since the other datetimelike indexes are also considered ``is_all_dates``.

#### Expected Output

``Out[5]:`` and ``Out[6]:`` should be the same as ``Out[3]:``.

#### Output of ``pd.show_versions()``

<details>

[paste the output of ``pd.show_versions()`` here below this line]

</details>

**Comment by jreback** [comment]
this should be made more like the other ``is_*`` methods (in base.py), where it simply checks the inferred type (and then checks for object), needs a doc-string

**Comment by gilbertoolimpio** [comment]
I'm a newbie here and I'm in search of fun! Can I help you with that? =)

**Comment by TomAugspurger** [comment]
@gilbertoolimpio yep, that'd be great. Do you know where to start?

**Comment by toobaz** [comment]
> this should be made more like the other is_* methods (in base.py), where it simply checks the inferred type (and then checks for object), needs a doc-string

I suggest then to just deprecate it, since it becomes equivalent to a simple ``isisntance`` call.

**Comment by jbrockmendel** [comment]
closed by #36697
